### ISSUE TITLE
Review customization point wording

### ISSUE THREAD

**Issue by CaseyCarter** [ericniebler/stl2#168]
...to ensure propagation of `constexpr` and `noexcept`.

**Comment by CaseyCarter** [comment]
`size` should not implicitly call `begin` on non-`Forward` ranges, since `begin` is not required to be equality-preserving. Doing so violates `SizedRange`'s semantic constraint that `size` is E-P until `begin` is called (so to speak).

In other words, this must be valid:
```c++
using namespace std::experimental::ranges;
template <class Range>
    requires InputRange<Range>() && SizedRange<Range>()
auto foo(Range&& r)
{
    assert(size(r) == size(r));
    return begin(r);
}
```
which it might not be if `size` returns `end(r) - begin(r)`, since `begin` would be called three times on the same input range.

**Comment by ericniebler** [comment]
Good point!

**Comment by CaseyCarter** [comment]
> `size` should not implicitly call `begin` on non-`Forward` ranges

...nor should `empty`. (`data` also calls `begin`, but only if it returns a pointer, which implies the range is `RandomAccess`.)

**Comment by ericniebler** [comment]
Fixed in #262. The issues with calling `begin` on an `InputRange` had been fixed previously.
